AK/AsyncBitStream.h

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2024, Dan Klishch <danilklishch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/AsyncStream.h>
+#include <AK/Debug.h>
+#include <AK/Endian.h>
+#include <AK/MaybeOwned.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/TemporaryChange.h>
+
+namespace AK {
+
+class AsyncInputLittleEndianBitStream;
+
+class BufferBitView {
+    // These are defined just to replace some 4s and 8s with meaningful expressions.
+    using WordType = u32;
+    using DoubleWordType = u64;
```

Comment:
No particular reason, `these are defined just to replace some 4s and 8s with meaningful expressions.` It is a bit nicer to think of reading in words and, therefore, chunks of `sizeof(Word)` bytes. As for `DoubleWordType`, I need to represent two consecutive words somewhere too.

In fact, `BufferBitView` works correctly if one replaces `WordType` with u64 and `DoubleWordType` with `unsigned __int128` (I haven't done this only because of the performance).